include/sys/timerfd.h

```diff
@@ -0,0 +1,80 @@
+/****************************************************************************
+ * include/sys/timerfd.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_SYS_TIMERFD_H
+#define __INCLUDE_SYS_TIMERFD_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdint.h>
+#include <fcntl.h>
+#include <time.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define TFD_NONBLOCK O_NONBLOCK
+#define TFD_CLOEXEC  O_CLOEXEC
+
+#define TFD_TIMER_ABSTIME TIMER_ABSTIME
+
+/* Get device minor number */
+
+#define TFD_FIOC_MINOR FIOC_MINOR
+
+/****************************************************************************
+ * Public Type Declarations
+ ****************************************************************************/
+
+/* Type for timer counter */
+
+typedef uint32_t timerfd_t;
```

Comment:
timerfd_t isn't used?

Reply:
It is used internally. Probably will move it to C-file.
I noticed that `eventfd` use `typedef uint32_t eventfd_t;` while Linux `typedef uint64_t eventfd_t;`, so I implemented `timerfd` also to use 32-bit.
@xiaoxiang781216 what do you think, should we move to `uint64_t` for both `eventfd` and `timerfd`?

Reply:
uint64_t doesn't always exist on all target, that's why uint32_t is used here. But, anyway uint32_t is enough in this case. BTW,  Linux use eventfd_t instead uint64_t which indicate that it expect to use the different type.

Reply:
Yeah, but unfortunately there is no such type for `timefd_t`.

Reply:
After reread eventfd and timerfd man, it look better to use uint64_t if possible. Since the value return from read may different from Linux, timerfd_t typedef is good to notice people there is some difference.

Reply:
Fix for eventfd is here: https://github.com/apache/incubator-nuttx/pull/5028/commits/a2e527dc13ded602484ecf9e7cb5089d4e8ac1bf

Reply:
@xiaoxiang781216 do you recommend to remove `timerfd_t` from the header? If yes, then interface in header will be not too clear IMO.

Reply:
No, I think it's better to keep in the public header file, so people get notice that the size may change from 64bits to 32bits.